core/node/consistency_checker/src/lib.rs

```diff
@@ -855,3 +804,24 @@ async fn wait_for_l1_batch_with_metadata(
             .ok();
     }
 }
+
+// Get settlement layer based on ETH tx in the database.
+pub async fn get_db_settlement_mode(
```

Comment:
Why is this a part of the consistency checker crate? It's not even used in it. Would it make sense to move to the call site (`node_framework`) and make private?